core/send.go

```diff
@@ -20,6 +20,15 @@ func SendCheckMsgs(chain Chain, msgs []types.Msg) bool {
 	return true
 }
 
+func SendMsgs(chain Chain, msgs []types.Msg) ([]MsgID, error) {
+	msgIDs, err := chain.SendMsgs(msgs)
+	if err != nil {
+		GetChainLogger(chain).Error("failed to send msgs", err, "msgs", msgs)
+		return nil, err
+	}
+	return msgIDs, nil
+}
```

Comment:
You should change `SendCheckMsgs` to return `([]MsgID, bool)` instead of introducting a new function.